support-frontend/assets/pages/contributions-checkout-in-epic/ContributionsCheckout.tsx

```diff
@@ -0,0 +1,166 @@
+		},
+		MONTHLY: {
+			...amounts.MONTHLY,
+			amounts: amounts.MONTHLY.amounts.slice(0, 2),
+		},
+		ANNUAL: {
+			...amounts.ANNUAL,
+			amounts: amounts.ANNUAL.amounts.slice(0, 2),
+		},
+	};
+};
+
+const mapStateToProps = (state: State) => ({
+	status: getStatus(state),
+	country: state.common.internationalisation.countryId,
+	countryGroupId: state.common.internationalisation.countryGroupId,
+	currency: state.common.internationalisation.currencyId,
+	contributionType: state.page.form.contributionType,
+	amounts: getAmounts(state),
+	selectedAmounts: state.page.form.selectedAmounts,
+	otherAmounts: state.page.form.formData.otherAmounts,
+});
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => {
+	return {
+		setSelectedContributionType: (contributionType: ContributionType) => {
+			dispatch(
+				updateContributionTypeAndPaymentMethod(contributionType, 'None'),
+			);
+		},
+		setSelectedAmount: (
+			amount: number | 'other',
+			contributionType: ContributionType,
+		) => {
+			dispatch(selectAmount(amount, contributionType));
+		},
+		setOtherAmount: (amount: string, contributionType: ContributionType) => {
+			dispatch(updateOtherAmount(amount, contributionType));
+		},
+	};
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type ContributionsCheckoutProps = ConnectedProps<typeof connector>;
+
+function ContributionsCheckout({
+	status,
+	country,
+	countryGroupId,
+	currency,
+	contributionType,
+	amounts,
+	selectedAmounts,
+	otherAmounts,
+	setSelectedContributionType,
+	setSelectedAmount,
+	setOtherAmount,
+}: ContributionsCheckoutProps): JSX.Element {
+	const ref = useOnHeightChangeEffect((height) => {
+		window.parent.postMessage({ type: 'IFRAME_HEIGHT', height }, '*');
+	});
+
+	return (
+		<div ref={ref} css={styles.container}>
+			{status === 'FORM' && (
+				<ContributionsCheckoutForm
+					country={country}
+					countryGroupId={countryGroupId}
+					currency={currency}
+					contributionType={contributionType}
+					amounts={amounts}
+					selectedAmounts={selectedAmounts}
+					otherAmounts={otherAmounts}
+					setSelectedContributionType={setSelectedContributionType}
+					setSelectedAmount={(amount) =>
+						setSelectedAmount(amount, contributionType)
+					}
+					setOtherAmount={(amount) => setOtherAmount(amount, contributionType)}
+				/>
+			)}
+
+			{status === 'SUBMITTING' && <ContributionsCheckoutSubmitting />}
+
+			{status === 'SUCCESS' && (
+				<ContributionsCheckoutThankYou
+					currency={currency}
+					contributionType={contributionType}
+					amount={getAmount(selectedAmounts, otherAmounts, contributionType)}
+				/>
+			)}
+		</div>
+	);
+}
+
+// ---- Hooks ---- //
+
+function useOnHeightChangeEffect(onChange: (height: number) => void) {
+	const { observe, height } = useDimensions({
+		useBorderBoxSize: true,
+		polyfill: ResizeObserver,
+	});
+
+	useEffect(() => {
+		onChange(height);
+	}, [height]);
+
+	return observe;
```

Comment:
Previously I had my own custom hook here. According to caniuse `ResizeObserver` has 91.32% which I don't think is high enough for us to use without polyfilling. It also has less support if you want the borderbox sizing. I started extending my custom hook to support optionally polyfilling when `ResizeObserver` isn't available but it became a bit of an async, `useRef`, `useCallback` nighmare. 

I've gone completely in the other direction currently: using a library for the hook & always polyfilling `ResizeObserver`. This has the downside of increasing page weight though. I'd be interested to hear thoughts on this one!